test/test_optim.py

```diff
@@ -340,9 +356,32 @@ def test_sparse_adam(self):
         self._test_rosenbrock_sparse(
             lambda params: optim.SparseAdam(params, lr=4e-2),
             [],
-            True
+            sparse_only=True,
+            test_cuda=False,
         )
-        with self.assertRaisesRegex(ValueError, "Invalid beta parameter at index 0: 1.0"):
+        self._test_rosenbrock_sparse(
+            lambda params: optim.SparseAdam(
+                params, lr=4e-2, fuse_kernels=False),
+            [],
+            sparse_only=True,
+            test_cuda=False,
+        )
+        if torch.cuda.is_available():
+            self._test_rosenbrock_sparse(
+                lambda params: optim.SparseAdam(params, lr=4e-2),
+                [],
+                sparse_only=True,
+                test_cuda=True,
+            )
+            self._test_rosenbrock_sparse(
```

Comment:
@BIT-silence rosenbrock tests for fused=True and fused=False are not enough. Can you add tests to make sure fused=True and fused=False are computing the same function. This way, if we get a PR that modifies the fused=False codepath, the tests would fail if fused=True codepath doesn't get changed